acquire/dynamic/windows/handles.py

```diff
@@ -0,0 +1,255 @@
+import csv
+import ctypes
+import gzip
+import io
+import struct
+import threading
+from logging import Filter, LogRecord, getLogger
+from queue import Queue
+from typing import Iterable, Optional
+
+from acquire.dynamic.windows.exceptions import OpenProcessError
+from acquire.dynamic.windows.ntdll import NtStatusCode, close_handle, ntdll
+from acquire.dynamic.windows.types import (
+    BOOL,
+    DWORD,
+    FILE_INFORMATION_CLASS,
+    HANDLE,
+    IO_STATUS_BLOCK,
+    OBJECT_INFORMATION_CLASS,
+    PHANDLE,
+    PUBLIC_OBJECT_TYPE_INFORMATION,
+    SYSTEM_HANDLE_INFORMATION_EX,
+    SYSTEM_HANDLE_TABLE_ENTRY_INFO_EX,
+    SYSTEM_INFORMATION_CLASS,
+    ErrorCode,
+    Handle,
+    ProcessAccess,
+)
+
+log = getLogger(__name__)
+
+advapi32 = ctypes.windll.advapi32
+advapi32.OpenProcessToken.argtypes = [HANDLE, DWORD, PHANDLE]
+
+kernel32 = ctypes.windll.kernel32
+kernel32.OpenProcess.restype = HANDLE
+kernel32.DuplicateHandle.argtypes = (HANDLE, HANDLE, HANDLE, ctypes.POINTER(HANDLE), DWORD, BOOL, DWORD)
+
+
+class DuplicateFilter(Filter):
+    def __init__(self) -> None:
+        super().__init__()
+        self.msgs = set()
+
+    def filter(self, record: LogRecord) -> bool:
+        seen = (msg := record.getMessage()) not in self.msgs
```

Comment:
I'm not a fan of using the `:=` operator like this. It is much quicker to parse the following as a human:
```suggestion
        msg =  record.getMessage()
        seen = msg not in self.msgs
```